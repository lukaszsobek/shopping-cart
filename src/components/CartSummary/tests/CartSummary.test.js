import { CartSummary } from "../";

describe("CartSummary",() => {


    it("calculates summary", () => {
        const props = [{
                tax: 5,
                net_price: 10
            },{
                tax: 5,
                net_price: 10
            },{
                tax: 10,
                net_price: 20
            }
        ];
        const testCartSummary = new CartSummary().calculateSummary(props);
        expect(testCartSummary.gross_total).toBe(43);
        expect(testCartSummary.net_total).toBe(40);
        expect(testCartSummary.tax_breakdown).toMatchObject({10: 2, 5: 1});
    });

    it("constructs summary",() => {
        const props = {10: 2, 5: 1}
        const testCartSummary = new CartSummary().constructTaxSummary(props);
        expect(testCartSummary).toMatchSnapshot();
    });

    it("renders component",() => {
        const props = {
            cartContent: [{
                tax: 5,
                net_price: 10
            },{
                tax: 5,
                net_price: 10
            },{
                tax: 10,
                net_price: 20
            }
        ]
        }
        const testCartSummary = new CartSummary(props).render();
        expect(testCartSummary).toMatchSnapshot();
    })
});