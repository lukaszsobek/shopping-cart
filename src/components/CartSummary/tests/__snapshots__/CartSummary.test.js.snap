// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CartSummary constructs summary 1`] = `
Array [
  <div
    className="tax-rate"
>
    <div
        className="label"
    >
        5
        %
    </div>
    <div
        className="value"
    >
        1
        €
    </div>
</div>,
  <div
    className="tax-rate"
>
    <div
        className="label"
    >
        10
        %
    </div>
    <div
        className="value"
    >
        2
        €
    </div>
</div>,
]
`;

exports[`CartSummary renders component 1`] = `
<div
  className="cart-summary"
>
  <div
    className="net-total"
  >
    <div
      className="label"
    >
      Net Total
    </div>
    <div
      className="value"
    >
      40
      €
    </div>
  </div>
  <div
    className="tax-rate"
  >
    <div
      className="label"
    >
      5
      %
    </div>
    <div
      className="value"
    >
      1
      €
    </div>
  </div>
  <div
    className="tax-rate"
  >
    <div
      className="label"
    >
      10
      %
    </div>
    <div
      className="value"
    >
      2
      €
    </div>
  </div>
  <div
    className="gross-total"
  >
    <div
      className="label"
    >
      Grand Total
    </div>
    <div
      className="value"
    >
      43
      €
    </div>
  </div>
</div>
`;
